UOP1_Project/Assets/Scripts/StateMachine/Utilities/InitOnlyAttribute.cs

```diff
@@ -0,0 +1,8 @@
+﻿using System;
+using UnityEngine;
+
+namespace UOP1.StateMachine
+{
+	[AttributeUsage(AttributeTargets.Field)]
+	public class InitOnlyAttribute : PropertyAttribute { }
```

Comment:
I think HelpBox should be shown on every SO, instead of being attached on fields via attribute

Reply:
I think I agree with @neon-age here! Also because one SO might have more than one of those fields. But then if you forget to add it to one of them it looks like that field is special.

Reply:
Hmmm, I thought it would be better to display specifically which fields had this behaviour. How should I reword it to make it more generic?

Reply:
Maybe custom editor for child classes that will look for runtime "data" field and displays the warning for it?
I can take that one if you don't know how to do that

Reply:
So, what you're saying is there should be a HelpBox on every field with this behaviour, but it should be added automatically instead of manually adding attributes on them?
You can take it if you want; I'm curious to know how you'd achieve this, though.

Reply:
A simpler way would be to always show it at the top, without iterating on fields, but then it would be visible on objects that have only shared variables, thus have no such behaviour

Reply:
That's exactly why I made it an attribute, it makes more sense to display it only where it applies. 

I still don't know what you're thinking with "iterating on fields", do you mean you'd show the HelpBox if a field in the SO has a matching field in the runtime component? I'd find that to be a little cryptic. What if someone names them differently? 

And speaking of naming things differently, the HelpBox wouldn't even be needed if we named these fields **InitialX**. So following my example in the thread, the field in the SO would be **InitialPosition**, while the runtime component would still have **Position**. I think just a naming convention like that would be pretty self-explanatory and intuitive for any designer.